File: my-leetcode-solutions/contains-duplicate.ts
// My Solution to LeetCode problem 217 - Contains Duplicate

// Brute Force Approach
function containsDuplicate(nums: number[]): boolean {
    // Sets check for duplicate values

    // Convert the array into a set
    const numsSet = new Set<number>();

    for (let i = 0; i < nums.length; i++) {
        numsSet.add(nums[i]);
    }


    // if numsSet.length == nums.length, return true, otherwise return false

    if(numsSet.size === nums.length) {
        return false
    } else { 
        return true
    }
};


// Time Complexity: O(n)

const testCases = [[1, 2, 3, 1], [], [1,1,1,1,1], [1,2,3,4]];

for(let i = 0; i < testCases.length; i++) {
    const output = containsDuplicate(testCases[i]);
    console.log(`Output: ${output} for input: ${testCases[i]}`);
}

export {}

File: my-leetcode-solutions-py/contains-duplicate.py
# My solution to 217 - Contains Duplicate

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        numsSet = {*nums}

        if len(numsSet) == len(nums):
            return False
        else: 
            return True


# Time Complexity:

test_cases = [
    [1, 2, 3, 1],
    [1, 2, 3, 4],
    [1, 1, 1, 3, 3, 4, 3, 2, 4, 2],
]

solution = Solution()

for i in range(len(test_cases)):
    output = solution.containsDuplicate(test_cases[i])
    print(f"Output: {output!r} for nums as {test_cases[i]!r}")
